src/cmd/ksh93/sh/args.c

```diff
@@ -441,25 +441,28 @@ int sh_argopts(int argc, char *argv[], void *context) {
             if (argc > 0) {
                 strsort(argv, argc, sortfn);
             } else if (np && (arp = nv_arrayptr(np)) && (args = nv_aivec(np, &bits))) {
-                struct Sort *sp;
                 char *cp;
                 int i, c, keys = 0;
-                size_t nodesize;
+
                 if (keylist) {
                     for (cp = keylist; (c = *cp); cp++) {
                         if (c == ',') keys++;
                     }
                     keys++;
                 } else {
-                    keylist = Empty;
```

Comment:
There are some places in code which directly compare against value of `Empty`. Have you verified such checks won't be affected by this change ?

Reply:
Yes, for this specific change. In general it looks like most, if not all, such uses were an "optimization" to avoid having lots of empty strings created (which hasn't been an issue for C compilers for at least three decades) or simply used as a silly stylistic idiom that obfuscates more than it illuminates. The only reason I haven't removed it everywhere is I haven't taken the time to verify it isn't a sneaky way to distinguish between a dynamically allocated string versus a static string for purposes of deciding if it can be passed to `free()`. Also, there is `EmptyStr` which is functionally identical. So I've been nibbling away at removing these references as I come across them individually, like this one, and am able to determine the transformation is safe.